worker/include/Utils.hpp

```diff
@@ -356,31 +356,29 @@ namespace Utils
 	template<typename T>
 	class ObjectPoolAllocator
 	{
-		thread_local static std::shared_ptr<std::vector<T*>> pool_data;
+		std::shared_ptr<std::vector<T*>> pool_data;
 
 	public:
 		typedef T value_type;
 		thread_local static Utils::ObjectPoolAllocator<T> Pool;
 
 		ObjectPoolAllocator()
 		{
-			if (!pool_data)
-			{
-				pool_data = std::shared_ptr<std::vector<T*>>(
-				  new std::vector<T*>(),
-				  [](std::vector<T*>* pool)
+			pool_data = std::shared_ptr<std::vector<T*>>(
+			  new std::vector<T*>(),
+			  [](std::vector<T*>* pool)
+			  {
+				  for (auto* ptr : *pool)
 				  {
-					  for (auto* ptr : *pool)
-					  {
-						  std::free(ptr);
-					  }
-					  delete pool;
-				  });
-			}
+					  std::free(ptr);
+				  }
+				  delete pool;
+			  });
 		}
 
 		template<typename U>
-		ObjectPoolAllocator(const ObjectPoolAllocator<U>& other) : ObjectPoolAllocator()
+		ObjectPoolAllocator(const ObjectPoolAllocator<U>& other)
+		  : pool_data(ObjectPoolAllocator<T>::Pool.pool_data)
 		{
```

Comment:
Do we actually need this constructor?

Reply:
@nazar-pc  It's constructor needed for ```allocator rebind```. Rebind needed for ```make_shared``` or ```allocate_shared```